gpjax/kernels/stationary/changepoint.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2023 The JaxGaussianProcesses Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+
+from dataclasses import dataclass
+
+from beartype.typing import (
+    Callable,
+    List,
+)
+import jax
+import jax.numpy as jnp
+from jaxtyping import Float
+import tensorflow_probability.substrates.jax.bijectors as tfb
+
+from gpjax.base import (
+    param_field,
+    static_field,
+)
+from gpjax.kernels.base import AbstractKernel
+from gpjax.typing import (
+    Array,
+    ScalarFloat,
+)
+
+
+@dataclass
+class ChangePoint(AbstractKernel):
+    r"""A change point kernel
```

Comment:
Is there a citation for your implementation of the kernel? We could add this to our `citation.py`. Also probably good to give a paper in the docstring. :)

Reply:
The idea started with the GPflow documentation, but I guess if you're thinking about a paper it would be Saatci et al 2010 : (https://icml.cc/Conferences/2010/papers/170.pdf). 

I'll add this in the doc strings.